lucene/core/src/test/org/apache/lucene/search/TestTopDocsCollector.java

```diff
@@ -179,6 +179,22 @@ public void testGetResultsFromStart() throws Exception {
     // get the last 5 only.
     assertEquals(5, tdc.topDocs(10).scoreDocs.length);
   }
+
+  public void testIllegalArguments() throws Exception {
+    final TopDocsCollector<ScoreDoc> tdc = doSearch(15);
+
+    IllegalArgumentException expected = expectThrows(IllegalArgumentException.class, () -> {
+      tdc.topDocs(20);
+    });
+
+    assertEquals(expected.getMessage(), "Illegal arguments when requesting top docs");
```

Comment:
nit: Junit assumes that the expected value is the first argument and the second value is what the test produced. Swapping both would produce better error messages.